src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health = 100;
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name) {
+    super(name);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name) {
+    super(name);
+  }
+  bite(obj) {
+    if (obj.hidden === false) {
```

Comment:
Consider adding a check to ensure that `obj` is an instance of `Herbivore` before accessing the `hidden` property. This will prevent potential errors if a non-Herbivore object is passed to the `bite` method.